woocommerce/woocommerce
# Product Collection: Fix: ensure the global product object is always ready for compatibility layer

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




This PR:
- ~adds a new method called `AbstractBlock::get_extra_block_settings()` allowing block to provide extra block settings that passed to `register_block_type()` function. If there are any of following properties are declared, they will be overridden in `AbstractBlock::register_block_type()`~:
   - ~`render_callback`~
   - ~`editor_script`~
   - ~`editor_style`~
   - ~`style`~
- ~updates the Product Template block to register with `skip_inner_blocks` flag. This is needs as the Product Template render the product loop, inner blocks shouldn't be rendered before the loop has been set up.~
- ~cleans up the unused imports in `AbstractBlock`.~
- filter the block settings of the Product Template block to pass in the `skip_inner_blocks` flag to disable default render routine of its inner blocks.

Closes #49598.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

This should be tested by developers only.

1. Add the following snippet to a block theme's functions.php

```php
add_action( 'woocommerce_after_shop_loop_item_title', function() {
   global $product;

   var_dump( $product->get_id() );
}, 10 );
```
2. Visit the shop page (ensure the shop page use the Product Collection block).
3. See the product ID printed.
4. Don't see any error/warning.


### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
